### DISCUSSION TITLE
Lost path to the maps

### DISCUSSION THREAD

**Discussion by Cinemannix** [devemux86/cruiser#325] [Q&A] [answered]
Hi,

on my Phone 1, since last update, it lost the path to the maps.
They are located on sd-card. After restart i have to switch to the second sdcard.
Before everything was fine.

Second phone, when i start brouter for download segements, it pops up an close. 

Phone 1 Android 8 second 9

Cine

**Comment by devemux86** [comment]
@Cinemannix thanks for the report.

What phones are these? Do they have 2 SD cards?

The storage folder can be selected in "Settings | Application | Advanced".
What do you have there? Perhaps you need to select again the SD card.

If SD card is not accessible when Cruiser starts it uses the internal storage.

---

Do you start BRouter from inside Cruiser or externally in Android apps?
Have you selected BRouter to use the SD card and the SD card was not accessible?

See also the images in:
- https://github.com/devemux86/cruiser/discussions/137

**Reply by Cinemannix** [reply]
Hi,

both are Blackview Outdoor phones 9500 / 9500 Plus

SD Card is accessible, because autosync works directly after start. it copy routes from drive to folder on SD.
I`m sure its since last update. I never had problems.

Second Phone

inside from cruiser. When as app directly, the in works.
on second phone is no sdcard


Cine

**Reply by devemux86** [reply]
What is selected in Cruiser "Settings | Application | Advanced | Storage folder"?

What storage is selected in BRouter?

**Reply by Cinemannix** [reply]
Phone 1 

/storage/09FD-1701/Android/media/gr.talent.cruiser

on this phone Brouter download works from cruiser, but first i have to switch from internal to external after reboot

**Reply by devemux86** [reply]
> but first i have to switch from internal to external after reboot

What do you mean? Switch Cruiser storage folder or something else?

**Reply by devemux86** [reply]
On the other phone, try to uninstall BRouter (and its data) and install it again.

**Reply by Cinemannix** [reply]
ok

**Comment by Cinemannix** [comment]
After reboot, cruise shows internal storage. sind update. It doesn`t remeber the sdcard

**Reply by Cinemannix** [reply]
i try to make some screenshots...later

**Reply by Cinemannix** [reply]
what is the different between  

/storage/09FD-1701/Android/media/gr.talent.cruiser
or
/storage/09FD-1701/Android/media/gr.talent.cruiser/files

**Reply by Cinemannix** [reply]
better to use the files the the root of gr.talent.cruiser

**Reply by devemux86** [reply]
The `/storage/09FD-1701/Android/media/gr.talent.cruiser` is the correct one.
The other one is not reported by Android.

**Reply by devemux86** [reply]
> After reboot, cruise shows internal storage. sind update. It doesn`t remeber the sdcard

When you restart Cruiser without rebooting the phone, does it remember the SD card?

If Cruiser starts and cannot access the SD card it reverts to the internal storage.

Perhaps you start Cruiser too soon after rebooting the phone
and the phone has not allowed SD card access to the apps yet?
Try some minutes later, what happens?

**Reply by Cinemannix** [reply]
but after reboot the app autosync starts first. then i start cruiser. afrer finishing sync

**Reply by devemux86** [reply]
Maybe the SD card is still locked, something seems wrong and cannot read / write the card.
Perhaps it needs more time, then start Cruiser.

If you start immediately a file manager, can it write on the SD card?

**Reply by Cinemannix** [reply]
yes

**Reply by devemux86** [reply]
> better to use the files the the root of gr.talent.cruiser

Where did you see the 2nd path? Android does not report such path.
Have you edited the files?

**Reply by Cinemannix** [reply]
<img src="https://github.com/devemux86/cruiser/assets/143206729/53344f32-7b7c-4fe4-8a55-e52eb43ec648" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/143206729/c020f2a6-077c-482e-b041-84f9d55b4d7e" width="50%"/>

**Reply by Cinemannix** [reply]
was a mistake data / media

**Reply by devemux86** [reply]
When you restart Cruiser without rebooting the phone, does it remember the SD card?

**Reply by devemux86** [reply]
And remove Cruiser from recent apps before restarting it.

**Reply by Cinemannix** [reply]
yes

**Reply by devemux86** [reply]
And in the other phone it works properly?
(same phone, different Android)

**Reply by Cinemannix** [reply]
now i switch off and wait for 5 minutes

**Reply by Cinemannix** [reply]
second phone i try later

**Reply by devemux86** [reply]
@Cinemannix thanks for the tests.

- Please restart Cruiser without rebooting the phones, but first remove Cruiser from recent apps
(and see if SD card is rememebered)

- And then start Cruiser after rebooting the phones

I will also check what else is possible to do. :slightly_smiling_face: 

---

About the BRouter, try to uninstall and reinstall the app.

Then do not start the BRouter app, but start directly BRouter downloader from inside Cruiser.

**Reply by Cinemannix** [reply]
so, after restart...now Autosync starts to check...
now i check with a filemanager...yes sdcarsd is here
now cruiser.... settings shows /storage/emulatet/0/Android/data/gr.talent.cruiser/files

**Reply by Cinemannix** [reply]
is there a logfile to send

**Reply by Cinemannix** [reply]
I`ll try the other folder on sd

copy all the files now...

**Reply by devemux86** [reply]
I will make an improvement to the storage folder.

A new Beta will be released very soon to test many new features.
It will contain this improvement, so you can test what happens to your phones.

**Reply by devemux86** [reply]
Until then, you can use the internal storage until it's fixed for good. :slightly_smiling_face:

**Reply by Cinemannix** [reply]
ok, but i test the other folder after copy

**Reply by devemux86** [reply]
Also please let me know about the BRouter tests, see my comment above: https://github.com/devemux86/cruiser/discussions/325#discussioncomment-7089843

(as BRouter is not related with Cruiser storage folder)

**Reply by Cinemannix** [reply]
brouter is ok now on the second phone

**Reply by Cinemannix** [reply]
lots of files to copy......

**Reply by Cinemannix** [reply]
ok, with the /storage/09FD-1701/Android/data/gr.talent.cruiser/files it works....  
I repeat the startup...

**Reply by Cinemannix** [reply]
yes, also after second reboot it work with the other folder on SD. 
I don`t know why, but for the moment it is ok...something changed in the rights maybe..

**Reply by devemux86** [reply]
This is a bit strange, the 2 folders in SD card to behave differently.

@Cinemannix many thanks for all the tests!

**Comment by devemux86** [comment]
@Cinemannix can you test again with:
- https://github.com/devemux86/cruiser/discussions/329

**Reply by Cinemannix** [reply]
At the moment I'm in croatia. If i have Wirelesslan I will Check...

**Reply by Cinemannix** [reply]
Seems to Work :-)

**Reply by devemux86** [reply]
@Cinemannix thanks for the report and the tests! :+1:

**Reply by Cinemannix** [reply]
Hi, today after starting the Phone, the right path was shown, but all the downloaded Maps were'nt active...
No problem for the moment.

**Reply by devemux86** [reply]
What map is displayed, the default online OpenStreetMap?
If the map files cannot be read, then the default online map is displayed.

Better reactivate the offline maps and then make tests with the new version.

**Comment by devemux86** [comment, marked as answer]
Should work properly in [Cruiser 3.0.34](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
